table/block_based/block_like_traits.h

```diff
@@ -21,10 +21,10 @@ template <typename T, CacheEntryRole R>
 Cache::CacheItemHelper* GetCacheItemHelperForRole();
 
 template <typename TBlocklike>
-Cache::CreateCallback GetCreateCallback(size_t read_amp_bytes_per_bit,
-                                        Statistics* statistics, bool using_zstd,
-                                        const FilterPolicy* filter_policy,
-                                        const TBlocklike& /*block*/) {
```

Comment:
Were you just using this block parameter to automatically instantiate the template? You can explicitly provide a type parameter on call without this block parameter. Wouldn't that avoid the extra stuff added in this PR?